sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -300,54 +301,44 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         "write files of Hive data source directly.")
     }
 
-    if (source == "memory") {
+    if (source == SOURCE_NAME_TABLE) {
+      assertNotPartitioned("table")
+
+      import df.sparkSession.sessionState.analyzer.CatalogAndIdentifier
+
+      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+      val CatalogAndIdentifier(catalog, identifier) = df.sparkSession.sessionState.sqlParser
+          .parseMultipartIdentifier(tableName)
+      val tableInstance = catalog.asTableCatalog.loadTable(identifier)
```

Comment:
Could you please elaborate? Which one do we want to handle here except DSv2 table?
(Personally I would say we should accelerate migration to DSv2 instead of spending efforts to extend supporting DSv1.)

Reply:
Ah sorry for the unclear, I mean the behavior when the table does not exist. Should we support create a new table and append data into? Seems a good-to-have feature. IMO it's the significant difference between the reader and the writer side.

Reply:
Ah OK. It brings additional options as we see in DataFrameWriterV2 to support create table, so if we want to do that, I think it warrants DataStreamWriterV2.